src/liballoc/raw_vec.rs

```diff
@@ -407,13 +407,34 @@ impl<T, A: AllocRef> RawVec<T, A> {
             return Err(CapacityOverflow);
         }
 
+        if needed_extra_capacity == 0 {
+            return Ok(());
+        }
```

Comment:
Is this check actually needed? `needs_to_grow` will return `false` if `needed_extra_capacity` is 0.

Reply:
True. I have replaced it with a debug assertion that it's non-zero.